provider/toolplugin.go

```diff
@@ -0,0 +1,23 @@
+package provider
```

Comment:
This should be in the `asdf` package IMHO. This is absolutely asdf-specific and won't be relevant for a second or third toolprovider implementation.